src/transformers/modeling_tf_utils.py

```diff
@@ -986,15 +988,15 @@ def load_tf_weights_from_safetensors(model, resolved_archive_file, ignore_mismat
     # Read the safetensors file
     with safe_open(resolved_archive_file, framework="tf") as safetensors_archive:
         mismatched_layers = []
-        weight_names = [format_weight_name(w.name, _prefix=_prefix) for w in model.weights]
+        weight_names = [strip_model_name_and_prefix(w.name, _prefix=_prefix) for w in model.weights]
```

Comment:
As: 
* The previous function `format_weight_name` added `_prefix`, whereas the new function `strip_model_name_and_prefix` removes `_prefix` if it's present. 
* `weight_names` are compared to those in `safetensors_archive`

Can we load in previously saved safetensors (before this PR) into our TF models?

Reply:
cc @Rocketknight1

Reply:
@amyeroberts I believe the previous code was just completely incorrect! Essentially, the relevant workflow (saving TF composite encoder-decoder models as safetensors and then reloading the checkpoint in TF) was not being tested, and actually didn't work because of the name prefix bug.

As such, I don't think there's a backwards compatibility issue here, because previous checkpoints weren't working at all. My suspicion is that not many people were saving encoder-decoder models in TF, and not many TF users were saving safetensors, and so the intersection of that venn diagram was tiny enough that no-one noticed the bug for a long time!

Reply:
Also, just to clarify: `_prefix` is almost always `None` when this function is called, in which case the behaviour is unchanged after this bugfix. `_prefix` is only defined when loading composite models like `EncoderDecoder`, which is the workflow that was broken before this.

Reply:
Thanks for explaining!